sound/soc/sof/core.c

```diff
@@ -582,17 +599,16 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 	/* initialize sof device */
 	sdev->dev = dev;
 
-	/* initialize default DSP power state */
-	sdev->dsp_power_state.state = SOF_DSP_PM_D0;
-
 	sdev->pdata = plat_data;
 	sdev->first_boot = true;
 	dev_set_drvdata(dev, sdev);
 
 	if (sof_core_debug)
 		dev_info(dev, "sof_debug value: %#x\n", sof_core_debug);
 
-	if (sof_debug_check_flag(SOF_DBG_DSPLESS_MODE)) {
+	/* select DSPless mode only when DSP OPS testing is not in use */
+	if (sof_debug_check_flag(SOF_DBG_DSPLESS_MODE) &&
+	    !sof_debug_check_flag(SOF_DBG_DSP_OPS_TEST_MODE)) {
```

Comment:
I would still inform the user that the DSPless mode is ignored because a DSP test mode is selected and taken precedence.

Reply:
i would really flip the logic and only take the DSP test mode when the user didn't explicitly bypass the DSP.